src/QMCWaveFunctions/Jastrow/ShortRangeCuspFunctor.h

```diff
@@ -0,0 +1,622 @@
+    OhmmsAttributeSet rAttrib;
+    rAttrib.add(radius, "rcut");
+    rAttrib.add(radius, "cutoff");
+    rAttrib.add(cusp_in, "cusp");
+    rAttrib.put(cur);
+
+    // set the hard cutoff radius if we have it
+    if (radius > 0.0)
+      cutoff_radius = radius;
+
+    // set the cusp if we have it
+    if (cusp_in > 0.0)
+      setCusp(cusp_in);
+
+    // loop over this node's children to read in variational parameter information
+    xmlNodePtr childPtr = cur->xmlChildrenNode;
+    while (childPtr != NULL)
+    {
+
+      // get the name of the child node
+      std::string cname((const char*)childPtr->name);
+      tolower(cname);
+
+      // read in a variable
+      if (cname == "var")
+      {
+
+        // read in the name of the variable
+        std::string v_name("string_not_set");
+        OhmmsAttributeSet att;
+        att.add(v_name, "name");
+        att.put(childPtr);
+
+        // read in the variable's info
+        if (v_name == "A")
+          set_variable_from_xml(PRE, childPtr, A, ID_A, Opt_A);
+        else if (v_name == "R0")
+          set_variable_from_xml(PRE, childPtr, R0, ID_R0, Opt_R0);
+        else if (v_name == "string_not_set")
+          PRE.error("variable name not set", true);
+        else
+          PRE.error("unrecognized variable name: " + v_name, true);
+
+      }
+
+      // read in the B coefficients
+      else if (cname == "coefficients")
+      {
+
+        // read in the id, whether to optimize, and the node type
+        std::string       id("string_not_set");
+        std::string     type("string_not_set");
+        std::string optimize("string_not_set");
+        OhmmsAttributeSet att;
+        att.add(      id,       "id");
+        att.add(    type,     "type");
+        att.add(optimize, "optimize");
+        att.put(childPtr);
+
+        // sanity check that this node has been specified as an array type
+        if (type != "Array")
+          PRE.error("ShortRangeCuspFunctor expected coefficients paramter array type to be \"Array\"", true);
+
+        // read in the vector of coefficients
+        putContent(B, childPtr);
+
+        // set the id if we have it
+        if ( id != "string_not_set" )
+          ID_B = id;
+
+        // if the coefficients are to be optimized, add them to the variable set
+        tolower(optimize);
+        if ( optimize == "yes" ) {
+          if ( id == "string_not_set" )
+            PRE.error("\"id\" must be set if we are going to optimize B coefficients", true);
+          Opt_B = true;
+          for (int i = 0; i < B.size(); i++) {
+            std::stringstream sstr;
+            sstr << ID_B << "_" << i;
+            myVars.insert(sstr.str(), B.at(i), Opt_B, optimize::OTHER_P);
+          }
+        } else if ( optimize == "no" ) {
+          Opt_B = false;
+        } else if ( optimize != "string_not_set" ) {
+          PRE.error("Unrecognized value for \"optimize\". Should be either yes or no", true);
+        }
+
+      }
```

Comment:
Add an abort here if the cname is unrecognized i.e. There is a typo.

Reply:
So far it is experimental, but in early testing it does a better job than using a short-ranged, finely spaced spline form.  As we learn more, the preferred functional form for these purposes may evolve...